### PR TITLE
Add source_in_ring args for rings API

### PR DESCRIPTION
**Pull request by zhoney**
[hugegraph/hugegraph#523](https://github.com/hugegraph/hugegraph/issues/523)

Change-Id: I8d7f148c072019946275419b00be059c433cf819

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/driver/HugeClient.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ private void initManagers(RestClient client, String graph) {
     private void checkServerApiVersion() {
         VersionUtil.Version apiVersion = VersionUtil.Version.of(
                                          this.version.getApiVersion());
-        VersionUtil.check(apiVersion, "0.38", "0.39",
+        VersionUtil.check(apiVersion, "0.39", "0.41",
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/HugeClient.java]: Keep compatible with low version (v0.10 must be compatible with v0.9)

File: src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java [hunk 2]
```diff
@@ -31,4 +39,27 @@ public RingsAPI(RestClient client, String graph) {
...
+        checkDegree(degree);
+        checkCapacity(capacity);
+        checkLimit(limit);
+
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("source", source);
+        params.put("direction", direction);
+        params.put("label", label);
+        params.put("max_depth", depth);
+        params.put("source_in_ring", sourceInRing);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java]: throw error if sourceInRing is true and api_version < 0.40

File: src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java [hunk 3]
```diff
@@ -31,4 +43,39 @@ public RingsAPI(RestClient client, String graph) {
...
+        checkLimit(limit);
+
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("source", source);
+        params.put("direction", direction);
+        params.put("label", label);
+        params.put("max_depth", depth);
+        if (sourceInRing) {
+            Version apiVersion = this.client.apiVersion();
+            if (apiVersion != null &&
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java]: move check of `apiVersion != null` to client.apiVersion()

File: src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java [hunk 4]
```diff
@@ -31,4 +43,39 @@ public RingsAPI(RestClient client, String graph) {
...
+                VersionUtil.match(apiVersion, "0.38", "0.40")) {
+                throw new ClientException("HugeGraphServer API version must " +
+                                          "be >= 0.40 to support " +
+                                          "source_in_ring arg of ring API, " +
+                                          "but current HugeGraphServer " +
+                                          "API version is: %s",
+                                          apiVersion.get());
+            }
+        }
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java]: move to line 56

File: src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java [hunk 5]
```diff
@@ -31,4 +43,39 @@ public RingsAPI(RestClient client, String graph) {
...
+
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("source", source);
+        params.put("direction", direction);
+        params.put("label", label);
+        params.put("max_depth", depth);
+        if (sourceInRing) {
+            Version apiVersion = this.client.apiVersion();
+            if (apiVersion != null &&
+                VersionUtil.match(apiVersion, "0.38", "0.40")) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java]: if(!VersionUtil.gte(apiVersion, "0.40"))

File: src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java (hunk 3 above)
Comment (zhoney) [reply src/main/java/com/baidu/hugegraph/api/traverser/RingsAPI.java]: apiVersoin is null if use RestClient directly instead of using HugeClient.

**Pull request merged**